streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1887,8 +1887,6 @@ private void updateSubscribedTopics(final Set<String> topics, final String logPr
         final Collection<String> existingTopics = subscriptionUpdates();
 
         if  (!existingTopics.equals(topics)) {
-            topics.addAll(existingTopics);
```

Comment:
This function is called by two callers: `xxxFromMetadata` and `xxxFromAssignment`. For the former we do not maintain the old topics but just replace with the passed in value, for the latter we still maintain the  old topics -- this it to take care if the leader did not assign all tasks / partitions due to assignment error. We should still keep that logic here.